src/main/scala/interfaces/Verification.scala

```diff
@@ -148,16 +149,55 @@ case class SiliconMappedCounterexample(
   val interpreter: ModelInterpreter[ExtractedModelEntry, Seq[ExtractedModelEntry]] = GenericDomainInterpreter(converter)
 
   override lazy val toString: String = {
-    val buf = converter.modelAtLabel
+    val extractedModel = converter.extractedModel
+    val buf_models = converter.modelAtLabel
       .map { case (label, model) => s"model at label $label:\n${interpret(model).toString}"}
       .mkString("\n")
-    s"$buf\non return: \n${interpret(converter.extractedModel).toString}"
+    val (buf_domains, buf_non_domain_functions) = functions_toString()
+    s"$buf_models\non return: \n${interpret(extractedModel).toString} $buf_domains $buf_non_domain_functions"
   }
   private def interpret(t: ExtractedModel): ExtractedModel =
     ExtractedModel(t.entries.map{ case (name, entry) => (name, interpret(entry)) })
   private def interpret(t: ExtractedModelEntry): ExtractedModelEntry = interpreter.interpret(t, Seq())
 
+  private def functions_toString() : (String, String) = {
+    //extractedModel.entries should be a bijection so we can invert the map
+    //and we only care about ref entries
+    val inv_entries = converter.extractedModel.entries.flatMap{ case (name, entry) => 
```

Comment:
For consistency with the rest of the codebase, all variables and functions should be `lowerCamelCase`, not `snake_case`. This applies to `inv_entries`, `sym_name`, `functions_toString`, etc etc.